controlplane/pkg/common/request_builder.go

```diff
@@ -0,0 +1,108 @@
+	if connectionID == "" {
+		connectionID = idGenerator()
+	}
+	return &networkservice.NetworkServiceRequest{
+		Connection: &connection.Connection{
+			Id:                     connectionID, // ID for NSE is managed by NSMgr
+			NetworkService:         requestConn.GetNetworkService(),
+			Context:                requestConn.GetContext(),
+			Labels:                 requestConn.GetLabels(),
+			NetworkServiceManagers: []string{nsmName},
+		},
+		MechanismPreferences: localMechanisms,
+	}
+}
+
+func createRemoteNSMRequest(connectionID, srcNsmName string, endpoint *registry.NSERegistration, remoteMechanisms []*connection.Mechanism, requestConn *connection.Connection) *networkservice.NetworkServiceRequest {
+	return &networkservice.NetworkServiceRequest{
+		Connection: &connection.Connection{
+			Id:                         connectionID,
+			NetworkService:             requestConn.GetNetworkService(),
+			Context:                    requestConn.GetContext(),
+			Labels:                     requestConn.GetLabels(),
+			NetworkServiceEndpointName: endpoint.GetNetworkServiceEndpoint().GetName(),
+			NetworkServiceManagers: []string{
+				srcNsmName, // src
+				endpoint.GetNetworkServiceManager().GetName(), // dst
+			},
+		},
+		MechanismPreferences: remoteMechanisms,
+	}
+}
+
+// Build request for the endpoint service in local service server
+func (builder *LocalRequestBuilder) Build(connectionID string, endpoint *registry.NSERegistration, fwd *model.Forwarder, requestConn *connection.Connection) *networkservice.NetworkServiceRequest {
+	if builder.nsmName == endpoint.GetNetworkServiceEndpoint().GetNetworkServiceManagerName() {
+		return createLocalNSERequest(connectionID, builder.idGenerator, builder.nsmName, fwd.LocalMechanisms, requestConn)
+	}
+	return createRemoteNSMRequest(connectionID, builder.nsmName, endpoint, fwd.RemoteMechanisms, requestConn)
+}
+
+// Build request for the endpoint service in remote service server
+func (builder *RemoteRequestBuilder) Build(connectionID string, _ *registry.NSERegistration, fwd *model.Forwarder,
+	requestConn *connection.Connection) *networkservice.NetworkServiceRequest {
+	return createLocalNSERequest(connectionID, builder.idGenerator, builder.nsmName, fwd.LocalMechanisms, requestConn)
+}
+
+// NewLocalRequestBuilder creates new request builder for the endpoint service in local service server
+func NewLocalRequestBuilder(m model.Model) *LocalRequestBuilder {
+	return &LocalRequestBuilder{
+		nsmName: m.GetNsm().GetName(),
+		idGenerator: func() string {
+			return m.ConnectionID()
+		},
+	}
+}
+
+// NewRemoteRequestBuilder creates new request builder for the endpoint service in remote service server
+func NewRemoteRequestBuilder(m model.Model) *RemoteRequestBuilder {
+	return &RemoteRequestBuilder{
+		nsmName: m.GetNsm().GetName(),
+		idGenerator: func() string {
```

Comment:
Can we use here `idGenerator: m.ConnecitonID` ?

Reply:
Fixed